### DISCUSSION TITLE
Intl.DateTimeFormat API doesn't calculate Persian leap years accurately

### DISCUSSION THREAD

**Discussion by Sinakhx** [nodejs/help#4935] [Q&A] [answered]
[This](https://github.com/nodejs/node/blob/6c86af9faa72ef68a728282a56f5d95d93ab3144/deps/icu-small/source/i18n/persncal.cpp#L110) is a very simplistic and inaccurate approach to calculating Jalali/Persian leap years.

I'm not sure if it's related to the nodejs work-group or the EcmaScript community, but as it's an implementation detail, I think it can be enhanced further, as the date calculations are very important.

I've actually reported this problem in an [other library's issues section](https://github.com/date-fns-jalali/date-fns-jalali/issues/62). Please follow along to see the details & examples [there](https://github.com/date-fns-jalali/date-fns-jalali/issues/62).

Please let me know if [This](https://github.com/nodejs/node/blob/6c86af9faa72ef68a728282a56f5d95d93ab3144/deps/icu-small/source/i18n/persncal.cpp#L110)  was an intentional approach or just due to insufficient astronomical research. Thanks in advance.

**Comment by targos** [comment]
@nodejs/i18n-api

**Reply by ryzokuken** [reply]
@Sinakhx this is an ICU issue, which is the library that this code actually comes from. The more I dig deeper into the ICU calendar code, the more problematic I find it, so I'm inclined to believe you're right here at once. I'm working on a long-term effort to fix many of these errors but it'll take a while before it gets to a point where Node.js does things correctly here.

**Comment by mhdawson** [comment]
@ryzokuken would it make sense to oint @Sinakhx to the upstream ICU repos/source?  I may not have enough context but my first thought is that is where the fix would need to go?

**Comment by srl295** [comment]
Don't fix in Node.js, please submit a ticket and PR to https://icu.unicode.org

**Comment by mhdawson** [comment]
@srl295 thanks for chiming in, was hoping we could point @Sinakhx to the right place upstream to fix.

**Comment by srl295** [comment, marked as answer]
yes, specifically:

@sinakhx  please open a PR against https://github.com/unicode-org/icu/blob/main/icu4c/source/i18n/persncal.cpp#L110-L115  you can link it here

**Reply by Sinakhx** [reply]
Thanks for your consideration & help.
I will open an issue there soon.

**Reply by ryzokuken** [reply]
Thanks @srl295

**Comment by Sinakhx** [comment]
I've created the pull request:
https://github.com/unicode-org/icu/pull/2077

Please consider the workflow in order for the PR to be merged & let me know about the details & upcoming steps.

**Comment by Sinakhx** [comment]
I guess my endeavor was in a wrong direction. So, I decided to close this issue:
https://github.com/unicode-org/icu/pull/2077#issuecomment-1127190385
